### PR TITLE
Update Autorest C#

### PR DESCRIPTION
**Pull request by pakrym**
Major features:

1. Ctor generation - 2 constructors are generated, default parameterless and one that would be used for deserialization later
2. Don't generate serializers for output only models and deserializers for input only.
3. Output only properties are readonly.
4. Newlines

**please double check the public API changes**

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/AnalyzeOperationResult_internal.cs [hunk 1]
```diff
@@ -12,13 +12,31 @@ namespace Azure.AI.FormRecognizer.Models
     /// <summary> Status and result of the queued analyze operation. </summary>
     internal partial class AnalyzeOperationResult_internal
     {
+        /// <summary> Initializes a new instance of AnalyzeOperationResult_internal. </summary>
+        internal AnalyzeOperationResult_internal()
```
Comment (pakrym) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/AnalyzeOperationResult_internal.cs]: internal is default for output models and public for input/input-output if parameterless ctor is defined in customization code non would be generated.

**Review by heaths — APPROVED** [review]
LGTM, but why generate the internal ctor with all the parameters? Is that easier for deserialization? With hand-rolled in KV, we just instantiated the model and set properties. I suppose it's all internal, though.

Also, thank you for adding some extra line breaks in the client code. Would be nice for readability to see that in the models, too. ;)

**Comment by pakrym** [comment]
> LGTM, but why generate the internal ctor with all the parameters? 

It's a requirement for struct based models so I decided to make it consistent this way.

**Review by tg-msft — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/AnalyzeOperationResult_internal.cs [hunk 2]
```diff
@@ -12,13 +12,31 @@ namespace Azure.AI.FormRecognizer.Models
...
+        {
+            Status = status;
+            CreatedDateTime = createdDateTime;
+            LastUpdatedDateTime = lastUpdatedDateTime;
+            AnalyzeResult = analyzeResult;
+        }
+
         /// <summary> Operation status. </summary>
-        public OperationStatus Status { get; set; }
+        public OperationStatus Status { get; internal set; }
```
Comment (annelo-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/AnalyzeOperationResult_internal.cs]: Do we need internal setters with the constructor that takes all the parameters?

**Review by annelo-msft — APPROVED** [review]
FR LGTM

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/AnalyzeOperationResult_internal.cs (hunk 2 above)
Comment (pakrym) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/AnalyzeOperationResult_internal.cs]: We don't use new constructors for deserialization yet.

Comment (annelo-msft) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/AnalyzeOperationResult_internal.cs]: Got it!

**Pull request merged**